test/gtest/core/test_reduce.cc

```diff
@@ -0,0 +1,247 @@
+
+        if (UCC_MEMORY_TYPE_HOST != mem_type) {
+            dsts = (typename T::type *)
+                    ucc_malloc(count * sizeof(typename T::type), "dsts buf");
+            EXPECT_NE(dsts, nullptr);
+            UCC_CHECK(ucc_mc_memcpy(dsts, ctxs[root]->args->dst.info.buffer,
+                                    count * sizeof(typename T::type),
+                                    UCC_MEMORY_TYPE_HOST, mem_type));
+        } else {
+            dsts = (typename T::type *)ctxs[root]->args->dst.info.buffer;
+        }
+        for (int i = 0; i < count; i++) {
+            typename T::type res =
+                    ((typename T::type *)((ctxs[0])->init_buf))[i];
+            for (int r = 1; r < ctxs.size(); r++) {
+                res = T::do_op(res,
+                              ((typename T::type *)((ctxs[r])->init_buf))[i]);
+            }
+            T::assert_equal(res, dsts[i]);
+        }
+        if (UCC_MEMORY_TYPE_HOST != mem_type) {
+            ucc_free(dsts);
+        }
+        return true;
+    }
+};
+
+TYPED_TEST_CASE(test_reduce, ReductionTypesOps);
+
+#define TEST_DECLARE(_mem_type, _inplace, _repeat)                             \
+    {                                                                          \
+        std::array<int, 3> counts{4, 256, 65536};                              \
+        for (int tid = 0; tid < UccJob::nStaticTeams; tid++) {                 \
+            for (int count : counts) {                                         \
+                UccTeam_h     team = UccJob::getStaticTeams()[tid];            \
+                int           size = team->procs.size();                       \
+                UccCollCtxVec ctxs;                                            \
+                this->set_mem_type(_mem_type);                                 \
+                this->set_inplace(_inplace);                                   \
+                this->data_init(size, TypeParam::dt, count, ctxs);             \
+                UccReq req(team, ctxs);                                        \
+                for (auto i = 0; i < _repeat; i++) {                           \
+                    req.start();                                               \
+                    req.wait();                                                \
+                    EXPECT_EQ(true, this->data_validate(ctxs));                \
+                    this->reset(ctxs);                                         \
+                }                                                              \
+                this->data_fini(ctxs);                                         \
+            }                                                                  \
+        }                                                                      \
+    }
+
+TYPED_TEST(test_reduce, single_host) {
+    TEST_DECLARE(UCC_MEMORY_TYPE_HOST, TEST_NO_INPLACE, 1);
+}
+
+TYPED_TEST(test_reduce, single_host_persistent)
```

Comment:
please keep "{" on the same line - make it uniform in all TYPED_TEST declarations